README.md

```diff
@@ -261,15 +261,8 @@ package managers:
 - `gem.txt`: run-time dependencies installed by gem.
 - `npm.txt`: run-time dependencies installed by npm.
 - `pip.txt`: a normal [pip `requirements.txt`][] file, for run-time
-  dependencies too.
-
-All of these files will install separately each line. See this example
```

Comment:
I did a test adding `pyodbc` and `sqlalchemy` in a `requirements.txt` file:

```
# pip install -r requirements.txt 
Collecting pyodbc (from -r requirements.txt (line 1))
  Using cached pyodbc-4.0.17.tar.gz
Collecting sqlalchemy (from -r requirements.txt (line 2))
Building wheels for collected packages: pyodbc
  Running setup.py bdist_wheel for pyodbc ... error
  Complete output from command /usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-Pdao7Y/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/tmpA4YaqFpip-wheel- --python-tag cp27:
  running bdist_wheel
  running build
  running build_ext
  building 'pyodbc' extension
  creating build
  creating build/temp.linux-x86_64-2.7
  creating build/temp.linux-x86_64-2.7/src
  gcc -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=4.0.17 -DSQL_WCHART_CONVERT=1 -I/usr/local/include/python2.7 -c src/pyodbcdbg.cpp -o build/temp.linux-x86_64-2.7/src/pyodbcdbg.o -Wno-write-strings
  cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
  In file included from src/pyodbcdbg.cpp:2:0:
  src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
  compilation terminated.
  error: command 'gcc' failed with exit status 1
  
  ----------------------------------------
  Failed building wheel for pyodbc
  Running setup.py clean for pyodbc
Failed to build pyodbc
Installing collected packages: pyodbc, sqlalchemy
  Running setup.py install for pyodbc ... error
    Complete output from command /usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-Pdao7Y/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-Oa3Ofm-record/install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_ext
    building 'pyodbc' extension
    creating build
    creating build/temp.linux-x86_64-2.7
    creating build/temp.linux-x86_64-2.7/src
    gcc -fno-strict-aliasing -g -O2 -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DPYODBC_VERSION=4.0.17 -DSQL_WCHART_CONVERT=1 -I/usr/local/include/python2.7 -c src/pyodbcdbg.cpp -o build/temp.linux-x86_64-2.7/src/pyodbcdbg.o -Wno-write-strings
    cc1plus: warning: command line option '-Wstrict-prototypes' is valid for C/ObjC but not for C++
    In file included from src/pyodbcdbg.cpp:2:0:
    src/pyodbc.h:56:17: fatal error: sql.h: No such file or directory
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/usr/local/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-Pdao7Y/pyodbc/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-Oa3Ofm-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-Pdao7Y/pyodbc/
```

The log is explicit enough IMHO.

Besides, with prior implementation there were the problems of slower installs (package managers already optimize it when dealing with multiple requirements at once) and build not being aborted on failure, as you can see in #53.

For corner cases, you can always add your custom `build.d` script, or add more lines to your `Dockerfile`, so I guess we should leave the corner cases out of the scope of the automatic dependency files and concentrate in most common ones.